### PR TITLE
Add support for Kommo

### PR DESCRIPTION
**Pull request by belyaev-dev**
If we switch from 'subdomain' to baseUrl, we can provide full url of platform to access both Kommo and amoCRM API

Solves each of these cases:
1. ru fork on subdomain.amocrm.ru
2. en fork on subdomain.kommo.com
3. en fork on subdomain.amocrm.com (deprecated)

### CODE REVIEW

**Comment by belyaev-dev** [comment]
It's breaking change btw

File: src/amo.ts [hunk 1]
```diff
@@ -76,12 +76,12 @@ export class Amo extends EventEmitter<WebhookEventMap> {
   private _file?: FileApi;
 
   constructor(
-    subdomain: string,
+    baseUrl: string,
```
Comment (shevernitskiy) [inline src/amo.ts:79]: property in snake_case pls

File: src/core/rest-client.ts [hunk 2]
```diff
@@ -12,11 +12,11 @@ export class RestClient {
   private _token?: OAuth;
 
   constructor(
-    private subdomain: string,
+    private baseUrl: string,
```
Comment (shevernitskiy) [inline src/core/rest-client.ts:15]: snake_case

File: src/amo.ts [hunk 3]
```diff
@@ -76,12 +76,12 @@ export class Amo extends EventEmitter<WebhookEventMap> {
...
     auth: OAuthCode | OAuth & Pick<OAuthRefresh, "client_id" | "client_secret" | "redirect_uri">,
     options?: Options,
   ) {
     super();
-    this.rest = new RestClient(subdomain, auth, options);
+    this.rest = new RestClient(baseUrl, auth, options);
```
Comment (shevernitskiy) [inline src/amo.ts:84]: snake_case

File: src/core/rest-client.ts [hunk 4]
```diff
@@ -12,11 +12,11 @@ export class RestClient {
...
     private auth: OAuthCode | OAuth & Pick<OAuthRefresh, "client_id" | "client_secret" | "redirect_uri">,
     private options?: Options,
   ) {
-    this.url_base = `https://${this.subdomain}.amocrm.ru`;
+    this.url_base = `https://${this.baseUrl}`; //subdomain.amocrm.ru || subdomain.kommo.com
```
Comment (shevernitskiy) [inline src/core/rest-client.ts:19]: snake_case

**Review by shevernitskiy — COMMENTED** [review]
Change all tests, examples and readme accroding to the introduces changes.
Reflects this whole moment in reame for user.

**Comment by belyaev-dev** [comment]
> Change all tests, examples and readme accroding to the introduced changes. Reflect this whole moment in readme for user.

Done

**Pull request merged**